### PR TITLE
Editor: evaluate from text rather than s-expr from rewrite-clj

### LINKED ISSUE #576: editor mode doesn't apply unquotes
Given a notebook with this:

```clojure
`(do ~'x)
```

In Clojure, this evaluates correctly to `(do x)`. But in Clerk's editor mode, I see the following:

image

Unquotes aren't correctly resolved.

### PR DESCRIPTION
**Pull request by borkdude**
Fixes #576

### CODE REVIEW

**Comment by borkdude** [comment]
Screenshot 2023-11-27 at 11 56 00

File: src/nextjournal/clerk/render/editor.cljs [hunk 1]
```diff
@@ -147,9 +147,14 @@
   (update doc :blocks (partial map (fn [{:as cell :keys [type text var form]}]
                                      (cond-> cell
                                        (= :code type)
-                                       (assoc :result                                                
-                                              {:nextjournal/value (cond->> (eval form)
-                                                                    var (hash-map :nextjournal.clerk/var-from-def))}))))))
+                                       (assoc :result {:nextjournal/value (cond->>
+                                                                              (sci/eval-string* (sci.ctx-store/get-ctx) text)
```
[the remark this answers is not in the document]
Comment (borkdude) [reply src/nextjournal/clerk/render/editor.cljs:151]: Yes we can, but we have to set `read-string` to a read-string that is coupled to the evaluation environment. If you're using SCI that would be `sci/parse-string`.

[the remark this answers is not in the document]
Comment (mk) [reply src/nextjournal/clerk/render/editor.cljs:151]: Yep, that should already be the case: https://github.com/nextjournal/clerk/blob/ad8e5a2c19aa55921ea357fa9005d563f80c53be/src/nextjournal/clerk/sci_env.cljs#L168

[the remark this answers is not in the document]
Comment (borkdude) [reply src/nextjournal/clerk/render/editor.cljs:151]: ok cool, I'll try it out

[the remark this answers is not in the document]
Comment (borkdude) [reply src/nextjournal/clerk/render/editor.cljs]: I've used `eval-string` now per Slack discussion.
We can make things more configurable when they need to.





[12:50](https://nextjournal.slack.com/archives/D02GWT6B4KB/p1701085844081119)
without being restricted to what cljs itself has

**Pull request merged**